fedimint-client/src/lib.rs

```diff
@@ -623,29 +623,35 @@ impl Drop for ClientArc {
                 }
 
                 let db = inner.db.clone();
-                let federation_id = inner.federation_id();
 
+                tokio::task::block_in_place(|| {
+                    tokio::runtime::Handle::current().block_on(async {
+                        // futures::executor::block_on(async {
+                        debug!(target: LOG_CLIENT, "Waiting for client task group to shut down");
+                        if let Err(err) = inner
+                            .task_group
+                            .clone()
+                            .shutdown_join_all(Some(Duration::from_secs(30)))
+                            .await
+                        {
+                            warn!(target: LOG_CLIENT, %err, "Error waiting for client task group to shut down");
+                        }
+                    });
+                });
```

Comment:
This might lead to a warning in the drop impl of `ExecutorInner` now, that calls `stop_executor`:

https://github.com/fedimint/fedimint/blob/3599ce3b8ab32667243a291271967fc090f11a87/fedimint-client/src/sm/executor.rs#L724-L726

Reply:
We can downgrade to `debug!` or rethink the approach if it becomes a problem, I guess.